### PR TITLE
Introduce @ExperimentalOkHttpApi

### PR DESCRIPTION
**Pull request by swankjesse**
I'm intending to use this to release OkHttp 5.0 before all of the new APIs are final.

### CODE REVIEW

**Comment by swankjesse** [comment]
(For a follow-up PR: annotating everything that I consider experimental!)

File: okhttp/src/main/kotlin/okhttp3/ExperimentalOkHttpApi.kt [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+  AnnotationTarget.FIELD,
+  AnnotationTarget.LOCAL_VARIABLE,
+  AnnotationTarget.VALUE_PARAMETER,
+  AnnotationTarget.CONSTRUCTOR,
+  AnnotationTarget.FUNCTION,
+  AnnotationTarget.PROPERTY_GETTER,
+  AnnotationTarget.PROPERTY_SETTER,
+  AnnotationTarget.TYPEALIAS
+)
+@RequiresOptIn(level = RequiresOptIn.Level.WARNING)
```
Comment (JakeWharton) [inline okhttp/src/main/kotlin/okhttp3/ExperimentalOkHttpApi.kt:43]: I'm tempted to say this should be error.

**Review by JakeWharton — APPROVED** [review]

File: okhttp/src/main/kotlin/okhttp3/ExperimentalOkHttpApi.kt [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+ * limitations under the License.
+ */
+package okhttp3
+
+/**
+ * Marks declarations that are experimental and subject to change without following SemVer
+ * conventions. Both binary and source-incompatible changes are possible, including complete removal
+ * of the experimental API.
+ *
+ * Do not use these APIs in modules that may be executed using a version of OkHttp different from
```
Comment (yschimke) [inline okhttp/src/main/kotlin/okhttp3/ExperimentalOkHttpApi.kt:23]: Do not use in published libraries without also marking experimental?

**Review by yschimke — APPROVED** [review]

File: okhttp/src/main/kotlin/okhttp3/ExperimentalOkHttpApi.kt (hunk 1 above)
Comment (swankjesse) [reply okhttp/src/main/kotlin/okhttp3/ExperimentalOkHttpApi.kt:43]: Done.

File: okhttp/src/main/kotlin/okhttp3/ExperimentalOkHttpApi.kt (hunk 2 above)
Comment (swankjesse) [reply okhttp/src/main/kotlin/okhttp3/ExperimentalOkHttpApi.kt:23]: Even better: Do not use these APIs in published libraries.

**Pull request merged**